### PR TITLE
rewrite last pr

### PR DESCRIPTION
**Pull request by MaikelStuivenberg**
## Summary by CodeRabbit

- **New Features**
	- Enhanced member addition process with new options for initial registration and navigation control.
	- Updated UI behavior to reflect changes in registration state and navigation capabilities.

- **Bug Fixes**
	- Improved control flow for navigation and button visibility in the member addition interface.

- **Documentation**
	- Updated method signatures to include new parameters for clarity.

### CODE REVIEW

File: lib/features/children/add_member/pages/add_member_counter_page.dart [hunk 1]
```diff
@@ -11,9 +11,10 @@ import 'package:givt_app/shared/models/analytics_event.dart';
 import 'package:givt_app/shared/widgets/fun_scaffold.dart';
 
 class AddMemberCounterPage extends StatefulWidget {
-  const AddMemberCounterPage({this.initialAmount, this.showTopUp = false, super.key});
+  const AddMemberCounterPage({this.initialAmount, this.showTopUp = false, this.initialRegistration = true, super.key});
```
Comment (TammiLion) [inline lib/features/children/add_member/pages/add_member_counter_page.dart:14]: I would not tie the naming to registration, canPop would be better for me.

**Review by TammiLion — APPROVED** [review]
Smol suggestion.

File: lib/features/children/add_member/pages/add_member_counter_page.dart (hunk 1 above)
Comment (MaikelStuivenberg) [reply lib/features/children/add_member/pages/add_member_counter_page.dart]: i did canPop + reversed the usage

File: lib/features/children/utils/add_member_util.dart [hunk 2]
```diff
@@ -5,12 +5,13 @@ import 'package:givt_app/features/children/add_member/pages/add_member_counter_p
 import 'package:givt_app/shared/widgets/extensions/route_extensions.dart';
 
 class AddMemberUtil {
-  static Future<void> addMemberPushPages(BuildContext context, {bool showTopUp = false}) async {
+  static Future<void> addMemberPushPages(BuildContext context, {bool showTopUp = false, bool initialRegistration = true}) async {
```
[the remark this answers is not in the document]
Comment (MaikelStuivenberg) [reply lib/features/children/utils/add_member_util.dart]: It's an optional param, so not missing

**Pull request merged**